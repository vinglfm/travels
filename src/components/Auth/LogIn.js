import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {logIn} from '../../actions';
import {connect} from 'react-redux';
import styles from './AuthModal.css';

class LogIn extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email:'',
            password:''
        };

        this.emailValidationSchema = {
            email: { type: 'email' }
        };

        this.passwordValidationSchema = {
            password: { type: 'string', min: 5, max: 12 }
        }

        this.logIn = this.logIn.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    logIn(event) {
        this.props.logIn(this.state);
        this.props.onClose();
        event.preventDefault();
    }

    handleChange(elem) {
        this.setState({[elem.target.name]: elem.target.value});
    }

    render() {
        return (
            <div>
                <div className={styles.modal__header}>
                    <a onClick={this.props.onBack} className={styles.modal__back}><i className='fas fa-arrow-left'></i></a>
                    <span>Log In</span>
                </div>
                <hr className={styles.modal__divider}/>
                <form className={styles.modal__form} name='LogInForm' onSubmit={this.logIn} noValidate>
                    <label className={styles.modal__form__label}>Email<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={styles.modal__form__input} name='email' type='email' onChange={this.handleChange}/>
                    <label className={styles.modal__form__label}>Password<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={styles.modal__form__input} name='password' type='password' placeholder='Password' onChange={this.handleChange}/>
                    <div className={styles.modal__form__group}>
                        <input className={styles.modal__form__group__submit} type='submit' value='Log In'/>
                    </div>
                </form>
            </div>
        );
    }
}

LogIn.propTypes = {
    onBack: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired
};

const mapDispatchToProps = {
    logIn
};

export default connect(null, mapDispatchToProps)(LogIn);