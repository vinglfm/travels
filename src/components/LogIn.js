import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Validator from 'fastest-validator';
import styles from './SignInModal.css';

const validator = new Validator();

export default class LogIn extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email:'',
            password:''
        };

        this.emailValidationSchema = {
            email: { type: 'email' }
        };

        this.passwordValidationSchema = {
            password: { type: 'string', min: 8, max: 16 }
        }

        this.logIn = this.logIn.bind(this);
        this.validateEmail = this.validateEmail.bind(this);
        this.validatePassword = this.validatePassword.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    logIn() {
        console.log(this.refs.email.value);
        console.log(this.refs.password.value);
    }

    handleChange(elem) {
        this.setState({[elem.target.name]: elem.target.value});
    }

    validateEmail() {
        return validator.validate(this.state, this.emailValidationSchema) === true;
    }

    validatePassword() {
        return validator.validate(this.state, this.passwordValidationSchema) === true;
    }

    render() {
        const isValidEmail = this.validateEmail();
        const isValidPassword = this.validatePassword();
        return (
            <div>
                <div className={styles.modal__header}>
                    <a onClick={this.props.baseComponent} className={styles.modal__back}><i className='fas fa-arrow-left'></i></a>
                    <span>Log In</span>
                </div>
                <hr className={styles.modal__divider}/>
                <form className={styles.modal__form} name='LogInForm' onSubmit={this.logIn} role='form'>
                    <label className={styles.modal__form__label}>Email<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={isValidEmail ? styles.modal__form__input : styles.modal__form__input__error} name='email' type='email' ref='email' onChange={this.handleChange}/>
                    <label className={styles.modal__form__label}>Password<span className={styles.modal__form__label__required}>*</span></label>
                    <input className={isValidPassword ? styles.modal__form__input : styles.modal__form__input__error} name='password' type='password' ref='password' placeholder='Password' onChange={this.handleChange}/>
                    <div className={styles.modal__form__group}>
                        <input className={styles.modal__form__group__submit} type='submit' disabled={!isValidEmail || !isValidPassword} value='Create Account'/>
                    </div>
                </form>
            </div>
        );
    }
}

LogIn.propTypes = {
    baseComponent: PropTypes.func.isRequired
};