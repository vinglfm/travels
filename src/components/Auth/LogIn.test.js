import enzyme, {shallow} from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new ReactSixteenAdapter() });

import React from 'react';
import LogIn from './LogIn';

it('renders LogIn', () => {
  const snapshot = shallow(<LogIn baseComponent={()=>{}} />);
  expect(snapshot).toMatchSnapshot();
});
