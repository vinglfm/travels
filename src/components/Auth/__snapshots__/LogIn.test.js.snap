// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LogIn 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogIn
    onBack={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
          <a
                    className={undefined}
                    onClick={[Function]}
          >
                    <i
                              className="fas fa-arrow-left"
                    />
          </a>
          <span>
                    Log In
          </span>
</div>,
        <hr
          className={undefined}
/>,
        <form
          className={undefined}
          name="LogInForm"
          noValidate={true}
          onSubmit={[Function]}
>
          <label
                    className={undefined}
          >
                    Email
                    <span
                              className={undefined}
                    >
                              *
                    </span>
          </label>
          <input
                    className={undefined}
                    name="email"
                    onChange={[Function]}
                    type="email"
          />
          <label
                    className={undefined}
          >
                    Password
                    <span
                              className={undefined}
                    >
                              *
                    </span>
          </label>
          <input
                    className={undefined}
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
          />
          <div
                    className={undefined}
          >
                    <input
                              className={undefined}
                              type="submit"
                              value="Create Account"
                    />
          </div>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className={undefined}
              onClick={[Function]}
>
              <i
                            className="fas fa-arrow-left"
              />
</a>,
            <span>
              Log In
</span>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-arrow-left"
/>,
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-arrow-left",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
            },
            "ref": null,
            "rendered": "Log In",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className={undefined}
>
              Email
              <span
                            className={undefined}
              >
                            *
              </span>
</label>,
            <input
              className={undefined}
              name="email"
              onChange={[Function]}
              type="email"
/>,
            <label
              className={undefined}
>
              Password
              <span
                            className={undefined}
              >
                            *
              </span>
</label>,
            <input
              className={undefined}
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
/>,
            <div
              className={undefined}
>
              <input
                            className={undefined}
                            type="submit"
                            value="Create Account"
              />
</div>,
          ],
          "className": undefined,
          "name": "LogInForm",
          "noValidate": true,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Email",
                <span
                  className={undefined}
>
                  *
</span>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              "Email",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "*",
                  "className": undefined,
                },
                "ref": null,
                "rendered": "*",
                "type": "span",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "name": "email",
              "onChange": [Function],
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Password",
                <span
                  className={undefined}
>
                  *
</span>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              "Password",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "*",
                  "className": undefined,
                },
                "ref": null,
                "rendered": "*",
                "type": "span",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className={undefined}
                type="submit"
                value="Create Account"
/>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "type": "submit",
                "value": "Create Account",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
            <a
                        className={undefined}
                        onClick={[Function]}
            >
                        <i
                                    className="fas fa-arrow-left"
                        />
            </a>
            <span>
                        Log In
            </span>
</div>,
          <hr
            className={undefined}
/>,
          <form
            className={undefined}
            name="LogInForm"
            noValidate={true}
            onSubmit={[Function]}
>
            <label
                        className={undefined}
            >
                        Email
                        <span
                                    className={undefined}
                        >
                                    *
                        </span>
            </label>
            <input
                        className={undefined}
                        name="email"
                        onChange={[Function]}
                        type="email"
            />
            <label
                        className={undefined}
            >
                        Password
                        <span
                                    className={undefined}
                        >
                                    *
                        </span>
            </label>
            <input
                        className={undefined}
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
            />
            <div
                        className={undefined}
            >
                        <input
                                    className={undefined}
                                    type="submit"
                                    value="Create Account"
                        />
            </div>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className={undefined}
                onClick={[Function]}
>
                <i
                                className="fas fa-arrow-left"
                />
</a>,
              <span>
                Log In
</span>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-arrow-left"
/>,
                "className": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-arrow-left",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
              },
              "ref": null,
              "rendered": "Log In",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className={undefined}
>
                Email
                <span
                                className={undefined}
                >
                                *
                </span>
</label>,
              <input
                className={undefined}
                name="email"
                onChange={[Function]}
                type="email"
/>,
              <label
                className={undefined}
>
                Password
                <span
                                className={undefined}
                >
                                *
                </span>
</label>,
              <input
                className={undefined}
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
/>,
              <div
                className={undefined}
>
                <input
                                className={undefined}
                                type="submit"
                                value="Create Account"
                />
</div>,
            ],
            "className": undefined,
            "name": "LogInForm",
            "noValidate": true,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Email",
                  <span
                    className={undefined}
>
                    *
</span>,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                "Email",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "*",
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": "*",
                  "type": "span",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "name": "email",
                "onChange": [Function],
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Password",
                  <span
                    className={undefined}
>
                    *
</span>,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                "Password",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "*",
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": "*",
                  "type": "span",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className={undefined}
                  type="submit"
                  value="Create Account"
/>,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": undefined,
                  "type": "submit",
                  "value": "Create Account",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
